zebra-network/src/protocol/external/codec.rs

```diff
@@ -45,6 +45,8 @@ pub struct Builder {
     version: Version,
     /// The maximum allowable message length.
     max_len: usize,
+    /// The address of the peer this codec is going to be used with.
+    addr: Option<String>,
```

Comment:
Since we're just using this as a metrics label, let's make the use clear in the name: 
```suggestion
    /// An optional label to use for reporting metrics.
    metrics_label: Option<String>,
```